src/components/main-view/main-view.jsx

```diff
@@ -1,37 +1,118 @@
 import { useState, useEffect } from "react";
 import { MovieCard } from "../movie-card/movie-card";
 import { MovieView } from "../movie-view/movie-view";
-
+import { LoginView } from "../login-view/login-view";
+import { SignupView } from "../signup-view/signup-view";
 
 export const MainView = () => {
-  const [movies, setMovies] = useState([]);
+    const storedUser = JSON.parse(localStorage.getItem("user"));
+    const storedToken = localStorage.getItem("token");
+    const [user, setUser] = useState(storedUser? storedUser : null);
+    const [token, setToken] = useState(storedToken? storedToken : null);
+    const [movies, setMovies] = useState([]);
+    const [selectedMovie, setSelectedMovie] = useState(null);
 
-  const [selectedMovie, setSelectedMovie] = useState(null);
+  // useEffect(() => {
+  //   if (!token) 
+  //     return;
+    
 
-  useEffect(() => {
-    fetch("https://movie-geeks-one.vercel.app/movies")
-      .then((response) => response.json())
-      .then((data) => {
-        console.log("movies from api:", data);
-        const moviesFromApi = data.map((movie) => {
-          return {
-            id: movie._id,
-            title: movie.Title,
-            genre: movie.Genre.Name,
-            image: movie.ImageURL,
-            director: movie.Director?.Name,
-            actors: movie.Actors,
-            description: movie.Description
-          };
-        });
+  //   fetch("https://movie-geeks-one.vercel.app/movies",
+  //   {
+  //     headers: {Authorization: `Bearer ${token}`}
+  //   })
+  //     .then((response) => response.json())
+  //     .then((moviesData) => {
+
+  //       const moviesFromApi = moviesData.map((movie) => {
+  //         return {
+  //           id: movie._id,
+  //           title: movie.Title,
+  //           genre: movie.Genre.Name,
+  //           image: movie.ImageURL,
+  //           director: movie.Director?.Name,
+  //           actors: movie.Actors,
+  //           description: movie.Description
+  //         };
+  //       });
+  //       console.log("moviesFromApi:", moviesFromApi);
+  //       setMovies(moviesFromApi);
+
+  //     });
+  // }, [token]);
 
+  useEffect(() => {
+    if (!token) return;
+  
+    fetch("https://movie-geeks-one.vercel.app/movies", {
+      headers: { Authorization: `Bearer ${token}` }
+    })
+      .then((response) => {
+        if (!response.ok) {
+          return response.text().then((text) => {
+            throw new Error(`Error ${response.status}: ${text}`);
```

Comment:
This is dangerous without properly handling this error. Once you throw there’s no where to go unless you catch it, which you’re not. Try returning a standardised error response that you can save and display to the user instead. Something like:

```
const [error, setError] = useState(undefined);

// in useEffect
 fetch(
  if (!response.ok) {
    setError(’No movies today, please try again’);
    return;
  }
).then((movieData)=>{
  if(! movieData) return;
  setError(undefined);
})
```

then somewhere in your UI you could have a nice error handling without destroying your app. .